### PR TITLE
Add CSRMonitorBundle to monitor CSR values

### PR DESCRIPTION
**Pull request by jchang0**
**Related issue**: 

Adding monitor bundle to monitor CSR registers (similar to coreBundle)

 feature request

**Impact**:  API addition (no impact on existing code)

**Development Phase**: implementation

**Release Notes**

### CODE REVIEW

File: src/main/scala/util/CoreMonitor.scala [hunk 1]
```diff
@@ -22,7 +22,12 @@ class CoreMonitorBundle(val xLen: Int) extends Bundle with Clocked {
   val inst = UInt(width = 32.W)
 }
 
+class CSRMonitorBundle(val xLen: Int) extends Bundle with Clocked {
+  val mie = UInt(width = xLen.W)
```
Comment (mwachs5) [inline src/main/scala/util/CoreMonitor.scala]: is there any reason you've picked  these two? Why not mstatus, etc?

**Comment by aswaterman** [comment]
You should probably add mstatus to the bundle, too, since that contains the global interrupt enable.

File: src/main/scala/util/CoreMonitor.scala (hunk 1 above)
Comment (jchang0) [reply src/main/scala/util/CoreMonitor.scala]: Those two are the one that @surajvellengar wanted

File: src/main/scala/subsystem/RocketSubsystem.scala [hunk 2]
```diff
@@ -55,7 +55,8 @@ trait HasRocketTiles extends HasTiles
   }
 
   def coreMonitorBundles = (rocketTiles map { t =>
-    t.module.core.rocketImpl.coreMonitorBundle
+    (t.module.core.rocketImpl.coreMonitorBundle,
+      t.module.core.rocketImpl.csr.csrMonitorBundle)
```
Comment (johningalls-sifive) [inline src/main/scala/subsystem/RocketSubsystem.scala]: I foresee problems mixing types here: **csr**MonitorBundle in **core**MonitorBundles

**Comment by jchang0** [comment]
Okay. I  did it from within federation only.  Going to hand that work off to Suraj

**Pull request closed without merging**